keras_trainer/trainer.py

```diff
@@ -107,15 +118,41 @@ def __init__(self, **options):
         )
 
         if not self.train_generator:
-            self.train_gen = self.train_data_generator.flow_from_directory(
-                self.train_dataset_dir,
-                batch_size=self.batch_size,
-                target_size=self.model_spec.target_size[:2],
-                class_mode='categorical'
-            )
-            self.num_classes = self.num_classes or self.train_gen.num_classes
-        else:
-            self.train_gen = self.train_generator
+            if self.train_dataset_dataframe is None and os.path.isdir(self.train_dataset_dir):
+                self.train_generator = self.train_data_generator.flow_from_directory(
+                    self.train_dataset_dir,
+                    iterator_mode=self.iterator_mode,
+                    n_outputs=self.n_outputs,
+                    batch_size=self.batch_size,
+                    target_size=self.model_spec.target_size[:2],
+                    class_mode='categorical'
+                )
+            elif self.train_dataset_dataframe is not None:
+                if isinstance(self.train_dataset_dataframe, pd.DataFrame):
+                    pass
+                elif self.train_dataset_dataframe.endswith('json'):
+                    self.train_dataset_dataframe = pd.read_json(self.train_dataset_dataframe)
+                elif self.train_dataset_dataframe.endswith('csv'):
+                    self.train_dataset_dataframe = pd.read_csv(self.train_dataset_dataframe)
+                else:
+                    raise ValueError('`train_dataset_dataframe` a json, a csv or a DataFrame object')
+                # We assume the dataframe will have the labels in 'class_probabilities' column and
+                # filenames under 'filename' column
+                self.train_generator = self.train_data_generator.flow_from_dataframe(
+                    self.train_dataset_dataframe,
+                    directory=self.train_dataset_dir,
+                    batch_size=self.batch_size,
+                    x_col="filename",
+                    y_col="class_probabilities",
+                    iterator_mode=self.iterator_mode,
+                    n_outputs=self.n_outputs,
+                    target_size=self.model_spec.target_size[:2],
+                    class_mode='probabilistic'
+                )
+            else:
+                raise ValueError('`train_dataset_dir` must be specified')
```

Comment:
Either `train_dataset_dir ` or `train_dataset_dataframe`, right? 
```suggestion
                raise ValueError('`train_dataset_dir` or `train_dataset_dataframe` must be specified.')
```